pyfar/dsp/dsp.py

```diff
@@ -1498,3 +1498,124 @@ def decibel(signal, domain='freq', log_prefix=None, log_reference=1,
         return log_prefix * np.log10(np.abs(data) / log_reference), log_prefix
     else:
         return log_prefix * np.log10(np.abs(data) / log_reference)
+
+
+def average(signal, mode='time', axis=None, phase_copy=None,
+            weights=None):
+    """
+    Average multichannel Signals.
+
+    Parameters
+    ----------
+    signal: Signal
+        Input signal of the Signal class
+    mode: string
+
+        ``'time'``
+            average in time domain. Note that this might cause artifacts if
+            the data is not aligned across channels.
+        ``'complex'``
+            average the complex spectra. Note that this might cause artifacts
+            if the data is not aligned across channels.
+        ``'magnitude'``
+            average the magnitude spectra and discard the phase
+            (see `phase_copy`)
+        ``'power'``
+            average the power spectra $|X|^2$ and discard the phase. The
+            squaring of the spectra is reversed before returning the averaged
+            signal.
+        ``'log_magnitude'``
+            average the log. magnitude spectra $20 \\log_{10}(X)$ and discard
+            the phase. The logarithm is reversed before returning the averaged
+            signal.
+
+        The default is ``'time'``
+    phase_copy: tuple
+        Some averaging modes discard the phase in which case the phase from
+        a channel before averaging can be copied to the averaged signal. Pass
+        the channel index as a tuple with as many elements as
+        ``signal.cshape``. The default ``None`` returns a zero-phase signal.
+    weights: array like
+        array that gives channel weights for averaging the data. Must be of
+        shape ``signal.cshape``. The default is ``None``, which applies equal
+        weights to all channels.
+
+    Returns
+    --------
+    averaged_signal: Signal
+        averaged input Signal
+
+    Notes
+    -----
+    The functions :py:func:`~pyfar.dsp.linear_phase` and
+    :py:func:`~pyfar.dsp.minimum_phase` can be used to obtain non-zero phase
+    responses. This can be usefull if the average mode discards the phase and
+    the paramter `phase_copy` can not be used.
+    """
+
+    # check input
+    if not isinstance(signal, pyfar.Signal):
+        raise TypeError('Input data has to be of type: Signal')
+
+    # set weights default
+    if weights is None:
+        weights = 1/(np.prod(signal.cshape))
+    else:
+        weights = weights/np.sum(weights)
```

Comment:
If we allow weights that can be broadcasted to cshape, we should not do the normalization. And add a note to the docstring that weights usually sum to 1 for most applications. Instead we could test if they can be braodcasted and raise a ValueError if not...